nexus/go-handler.md

```diff
@@ -0,0 +1,307 @@
+
+**Goals**
+
+- Expose a simple, high level API for binding Nexus operations to common Temporal constructs
+- Support for starting and canceling operations
+- Support multi-tenant applications with minimal boilerplate
+
+**Non-Goals**
+
+- Support for getting an operation's result or info
+- API definitions for other SDKs
+- API for invoking Nexus calls - refer to the [Nexus Go SDK][nexus-go-sdk] for an out-of-workflow reference client
+- Server implementation details
+- Worker / server API contract
+- Allow advanced users to implement Nexus inbound and outbound request proxies for cross-organization calls where
+  data converters are not shared and data is not exposed to Temporal server.
+
+  This will quickly become a requirement for production use cases, especially with Temporal Cloud's zero trust model. It
+  will be addressed shortly after Nexus becomes available for preview.
+
+> NOTE: These are non-goals for _this_ proposal, but they will be defined in follow up proposals / PRs.
+
+## Background
+
+Nexus is a thin protocol on top of HTTP, defined in [this spec][nexus-http-api]. Temporal serves as an HTTP proxy,
+translating Nexus requests into tasks that can be polled on by an SDK worker on its configured task queue.
+Due to the synchronous nature of HTTP, the generated tasks are not persisted, and require an available worker
+to be processed.
+
+The process of translating requests into tasks and delivering them to workers is out of scope for this proposal.
+
+## SDK Additions
+
+The Temporal Go SDK exposes a `Worker` interface that embeds a `Registry` interface defined as:
+
+```go
+type Registry interface {
+	WorkflowRegistry
+	ActivityRegistry
+}
+```
+
+Add and embed a third registry type: `OperationRegistry` with the following method:
+
+```go
+type OperationRegistry interface {
+	RegisterOperation(operation string, handler operation.Handler)
+	// Alternative: get the operation name from handler.Name()
+	// RegisterOperation(handler operation.Handler)
+}
+```
+
+`operation.Handler` is an interface with a minimal definition of:
+
+```go
+package operation // go.temporal.io/sdk/operation
+
+type Handler interface {
+	StartOperation(context.Context, *StartOperationRequest) (StartOperationResponse, error)
+	CancelOperation(context.Context, *CancelOperationRequest) error
+	// Later we will add GetResult and GetInfo as well
+}
+
+// Maps almost 1:1 to the Nexus Go SDK StartOperationRequest, exposing HTTP header and body directly instead of the HTTP
+// Request object.
+type StartOperationRequest struct {
+	Operation   string
+	RequestID   string
+	CallbackURL string
+
+	URL    *url.URL
+	Header http.Header
+	Body   []byte
+}
+
+// Maps almost 1:1 to the Nexus Go SDK CancelOperationRequest, exposing HTTP header and body directly instead of the
+// HTTP Request object.
+type CancelOperationRequest struct {
+	Operation   string
+	OperationID string
+
+	URL    *url.URL
+	Header http.Header
+}
+
+type OperationResponse interface {
```

Comment:
You called this `StartOperationResponse` above

Reply:
Thanks, that's the downside of markdown, you don't get static analysis. I'll fix above to match the nexus API.